File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    //Ashwini
    int maxSubArray(vector<int>& nums) {
        int n= nums.size();
        int sum=nums[0];
        int max_n=nums[0];

        for (int i=1; i<n; i++){
           sum= max(nums[i] , sum + nums[i]);
           max_n= max( sum , max_n);
            
        }

        return max_n;
        
    }
};

File: 53-maximum-subarray/maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        s=0
        max_sum= float('-inf')
        n= len(nums)

        for i in range (n):
            s+= nums[i]

            max_sum= max (s, max_sum)
            
            if (s<0):
                s=0

        return max_sum
        
                